wgpu-core/src/device/mod.rs

```diff
@@ -1086,10 +1093,14 @@ impl<A: HalApi> Device<A> {
         }
         let format = desc.format.unwrap_or(texture.desc.format);
         if format != texture.desc.format {
-            return Err(resource::CreateTextureViewError::FormatReinterpretation {
-                texture: texture.desc.format,
-                view: format,
-            });
+            if texture.desc.view_formats.contains(&format) {
+                self.require_downlevel_flags(wgt::DownlevelFlags::VIEW_FORMATS)?;
+            } else {
+                return Err(resource::CreateTextureViewError::FormatReinterpretation {
+                    texture: texture.desc.format,
+                    view: format,
+                });
+            }
```

Comment:
This is a hyper-nit:

This would read better inverted:

```suggestion
            if !texture.desc.view_formats.contains(&format) {
                return Err(resource::CreateTextureViewError::FormatReinterpretation {
                    texture: texture.desc.format,
                    view: format,
                });
            }
            self.require_downlevel_flags(wgt::DownlevelFlags::VIEW_FORMATS)?;
```

As the requiring of the downlevel flags is actually unrelated to if the view format is in the supported list.